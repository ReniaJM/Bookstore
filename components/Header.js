import React, { Component } from 'react';
import {Link} from "react-browser-router";

class Header extends Component {
    constructor(){
        super()
        this.state= {
            bookstorename: 'black book',
            clicked: true,
            textColour: 'black',
            backgroundColor: 'white'
        }
    }
        handleClick=() => {
        if(this.state.clicked){
            this.setState({
                bookstorename: 'white book',
                textColour: 'white',
                backgroundColor: 'black'
            })
        }else{
            this.setState({
                bookstorename: 'black book',
                textColour: 'black',
                backgroundColor: 'white'
            })
        }
        this.setState({
            clicked: !this.state.clicked,
        })
}
    render(){
        let headercss={
            colour: this.state.textColour,
            backgroundColor: this.state.backgroundColor
        }
        return (
            <div className ='row header' style={headercss} onClick={this.handleClick}>
               <h2>{this.state.bookstorename}</h2>
                <Link to ='/admin'><button className='btn btn-info gotoAdmin'>Admin Panel</button></Link>
            </div>
        );
    }
}

export default Header