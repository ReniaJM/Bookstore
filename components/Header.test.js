import React from 'react';
import ReactDOM from 'react-dom';
import {configure, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Header from "./Header";

configure({adapter: new Adapter()})
describe('header test', () =>{
    it('header renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Header />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('header changes text according to state', () =>{
        const wrapper =shallow (<Header/>)
        console.log(wrapper.debug())
        expect(wrapper.find('h2').text()).toBe('black book')
        wrapper.setState({
            bookstorename:'white book'
        })
        expect(wrapper.find('h2').text()).toBe('white book')
    })
    it('header state changes after clicked on div',() =>{
        const wrapper =shallow (<Header/>)
        expect(wrapper.state().bookstorename).toBe('black book')
        wrapper.find('.header').simulate('click')
        expect(wrapper.state().bookstorename).toBe('white book')
    })
    it('snapshot matches', () =>{
        const wrapper =shallow (<Header/>)
        expect(wrapper).toMatchSnapshot();
    })
})