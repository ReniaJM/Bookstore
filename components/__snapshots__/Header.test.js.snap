// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header test snapshot matches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h2>
        black book
      </h2>,
      "className": "row header",
      "onClick": [Function],
      "style": Object {
        "backgroundColor": "white",
        "colour": "black",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "black book",
      },
      "ref": null,
      "rendered": "black book",
      "type": "h2",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2>
          black book
        </h2>,
        "className": "row header",
        "onClick": [Function],
        "style": Object {
          "backgroundColor": "white",
          "colour": "black",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "black book",
        },
        "ref": null,
        "rendered": "black book",
        "type": "h2",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
